wideq/core.py

```diff
@@ -397,3 +411,59 @@ def get_device_config(self, device_id, key, category='Config'):
             'data': '',
         })
         return res['returnData']
+
+    def delete_permission(self, device_id):
+        """ for using mobile app and HA at the same time"""
+        self.post('rti/delControlPermission', {
+            'deviceId': device_id,
+        })
+
+    def get_power_data(self, device_id, period):
+        """ get AC power consumption data"""
+        res = self.post('aircon/inquiryPowerData', {
+            'deviceId': device_id,
+            'period': period,
+        })
+        code = res.get('returnCd')  # returnCode can be missing.
+        if code == '0000':
+            return res['powerData']
+        elif code == '0010':
+            return '0'
+        else:
+            raise MonitorError(device_id, code)
+
+    def get_water_usage(self, device_id, typeCode, sDate, eDate):
+        """get water purifier usage data"""
+        res = self.post('rms/inquiryWaterConsumptionInfo', {
+            'deviceId': device_id,
+            'type': typeCode,
+            'startDate': sDate,
+            'endDate': eDate,
+        })
+        code = res.get('returnCd')  # returnCode can be missing.
+        if code != '0000':
+            raise MonitorError(device_id, code)
+        else:
+            return res['item']
+            
+    def get_outdoor_weather(self, area):
+        """ get outdoor weather data from accuweather.
+        it need area code"""
```

Comment:
Interesting! Can you clarify what an "area code" is exactly and how to obtain one?